### PR TITLE
fix(predicate): add special tag key, fixed children number

### PR DESCRIPTION
**Pull request by kelwang**
Closes https://github.com/influxdata/influxdb/issues/15677

add _measurement and _field special tag key.
fixed a logical node issue.
manually tested as well.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: predicate/logical.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func (op LogicalOperator) Value() (datatypes.Node_Logical, error) {
 // LogicalNode is a node type includes a logical expression with other nodes.
 type LogicalNode struct {
 	Operator LogicalOperator `json:"operator"`
-	Children []Node          `json:"children"`
+	Children [2]Node         `json:"children"`
```
Comment (e-dard) [inline predicate/logical.go]: What warranted this change?

File: predicate/parser.go [hunk 2]
```diff
@@ -91,9 +89,29 @@ func (p *parser) parseLogicalNode() (Node, error) {
...
 			}
-			n.Children = append(n.Children, tr)
+			if n.Children[0] == nil {
+				n.Children[0] = tr
+			} else {
+				n.Children[1] = tr
+			}
 		case influxql.AND:
 			n.Operator = LogicalAnd
+			if n.Children[1] != nil {
```
Comment (jsteenb2) [inline predicate/parser.go]: nit: could flip the comparison here and just continue, unindenting the main logic of this block

```go
if n.Children[1] == nil {
  continue
}
other stuffs...
```

File: predicate/parser.go [hunk 3]
```diff
@@ -201,3 +223,13 @@ scanRegularTagValue:
 		}
 	}
 }
+
+// peekRune returns the next rune that would be read by the scanner.
+func (p *parser) peekTok() influxql.Token {
+	tok, _, _ := p.scanIgnoreWhitespace()
+	if tok != influxql.EOF {
```
Comment (jsteenb2) [inline predicate/parser.go]: curious why there is no unscan when its `EOF`?

File: predicate/logical.go (hunk 1 above)
Comment (jsteenb2) [reply predicate/logical.go]: had same question, is this to avoid the panic on zero value slice(`nil`)?

Comment (kelwang) [reply predicate/logical.go]: this line got me, didn't saw it in the previous commit
https://github.com/influxdata/influxdb/blob/68cfd386c377d1869dd54805caefdd0c08c4860b/tsdb/tsm1/predicate.go#L163

File: predicate/parser.go (hunk 3 above)
Comment (kelwang) [reply predicate/parser.go]: the scan will consume the current position until eof.

File: predicate/parser.go (hunk 2 above)
Comment (kelwang) [reply predicate/parser.go]: good idea

**Pull request merged**